podman_compose.py

```diff
@@ -3585,29 +3613,135 @@ async def compose_logs(compose: PodmanCompose, args: argparse.Namespace) -> None
+        targets.extend(containers)
+        for container in containers:
+            service_by_container[container] = service
+
+    should_use_colors = (
+        (len(args.services) > 1 or args.names)
+        and not args.latest
+        and sys.stdout.isatty()
+        and not getattr(args, "no_color", False)
+    )
+
+    if should_use_colors:
+        max_service_length = 0
+        for target in targets:
+            cnt = compose.container_by_name[target]
+            service_name = cnt["_service"]
+            container_name = cnt["name"]
+
+            if getattr(args, 'names', False):
+                expected_name = compose.format_name(service_name, str(cnt["num"]))
+
+                if container_name == expected_name:
+                    display_name = compose.join_name_parts(service_name, str(cnt["num"]))
+                else:
+                    display_name = container_name
+            else:
+                display_name = container_name
+
+            curr_length = len(display_name)
+            max_service_length = (
+                curr_length if curr_length > max_service_length else max_service_length
+            )
+
+        tasks = []
+        service_colors: dict[str, str] = {}
+
+        for target in targets:
+            cnt = compose.container_by_name[target]
+            service_name = cnt["_service"]
+            container_name = cnt["name"]
+
+            if getattr(args, 'names', False):
+                expected_name = compose.format_name(service_name, str(cnt["num"]))
+                if container_name == expected_name:
+                    display_name = compose.join_name_parts(service_name, str(cnt["num"]))
+                else:
+                    display_name = container_name
+            else:
+                display_name = container_name
+
+            if service_name not in service_colors:
+                color_idx = len(service_colors) % len(compose.console_colors)
+                service_colors[service_name] = compose.console_colors[color_idx]
+
+            color = service_colors[service_name]
+
+            space_suffix = " " * (max_service_length + 1 - len(display_name))
+            log_formatter = "{}{}{}|\x1b[0m".format(color, display_name, space_suffix)
+
+            podman_args = []
+            if args.follow:
+                podman_args.append("-f")
+            if args.names:
+                podman_args.append("-n")
+            if args.since:
+                podman_args.extend(["--since", args.since])
+            if args.tail and args.tail != "all":
+                podman_args.extend(["--tail", args.tail])
+            if args.timestamps:
+                podman_args.append("-t")
+            if args.until:
+                podman_args.extend(["--until", args.until])
+            podman_args.append(target)
+
+            task = asyncio.create_task(
+                compose.podman.run([], "logs", podman_args, log_formatter=log_formatter),
```

Comment:
Future improvement (not this PR): run only single `podman logs` command and improve the formatter to be able to apply different formatting depending on container ID that podman logs emits on every line.